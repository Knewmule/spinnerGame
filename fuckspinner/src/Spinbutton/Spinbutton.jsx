import {styled} from 'styled-components';

const Spin = styled.div`
display:flex;
flex-direction:column;
align-items:center;
`
const Now = styled.button`
display:flex;
justify-content:space-evenly;
align-items:flex-end;
margin:0px;
width:100%;
height:100px;
border-bottom:1px solid black;
font-size:30px;
background-color: #00ff00;
border-top-left-radius:106px;
border-top-right-radius:110px;
color:black;
`
const Score = styled.p`
font-size:28px;
color:black;
`
export default function Spinbutton({spin}){

    return(
        <Spin>
            <Now onClick={spin}>Spin</Now>
            <Score>100</Score>
        </Spin>
    )
    
}