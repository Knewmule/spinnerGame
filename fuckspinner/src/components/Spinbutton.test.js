import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import {Spinbutton,Now,Score, Spin} from './Spinbutton';
// import toJson from 'enzyme-to-json'
describe("Spinbutton Component", () => {
  it("describes Spin style", () =>{
    const spin = renderer.create(<Spin />).toJSON()
    expect(spin).toHaveStyleRule('display','flex')
    expect(spin).toHaveStyleRule('flex-direction','column')
    expect(spin).toHaveStyleRule('align-items','center')
    expect(spin).toMatchSnapshot()
  })
  it("describes Now style", () =>{
    const now = renderer.create(<Now />).toJSON()
    expect(now).toHaveStyleRule('display','flex')
    expect(now).toHaveStyleRule('justify-content','space-evenly')
    expect(now).toHaveStyleRule('align-items','flex-end')
    expect(now).toHaveStyleRule('margin','0px')
    expect(now).toHaveStyleRule('width','100%')
    expect(now).toHaveStyleRule('height','100px')
    expect(now).toHaveStyleRule('border-bottom','1px solid black')
    expect(now).toHaveStyleRule('font-size','30px')
    expect(now).toHaveStyleRule('background-color','#00ff00')
    expect(now).toHaveStyleRule('border-top-left-radius','106px')
    expect(now).toHaveStyleRule('border-top-right-radius','110px')
    expect(now).toHaveStyleRule('color','black')
    expect(now).toMatchSnapshot()
  })
  it("describes Score style", () =>{
    const score = renderer.create(<Score />).toJSON()
    expect(score).toHaveStyleRule('font-size','28px')
    expect(score).toHaveStyleRule('color','black')
    expect(score).toMatchSnapshot()
  })
});