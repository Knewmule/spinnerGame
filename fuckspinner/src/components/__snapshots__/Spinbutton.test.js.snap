// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinbutton Component describes Now style 1`] = `
.c0 {
  display: flex;
  justify-content: space-evenly;
  align-items: flex-end;
  margin: 0px;
  width: 100%;
  height: 100px;
  border-bottom: 1px solid black;
  font-size: 30px;
  background-color: #00ff00;
  border-top-left-radius: 106px;
  border-top-right-radius: 110px;
  color: black;
}

<button
  className="c0"
/>
`;

exports[`Spinbutton Component describes Score style 1`] = `
.c0 {
  font-size: 28px;
  color: black;
}

<p
  className="c0"
/>
`;

exports[`Spinbutton Component describes Spin style 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  align-items: center;
}

<div
  className="c0"
/>
`;
